### PR TITLE
chore(ci): added a ci check for driver schema and api version.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area CI

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
> Please double check driver/SCHEMA_VERSION file.

It works :rofl: I want to find a way to tag pr creator though!

**Comment by FedeDP** [comment]
THe message will be added just once for each PR, not on every commit: 
>     allow-repeats: false

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -391,3 +404,29 @@ jobs:
...
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: write
+    needs: paths-filter
+    if: needs.paths-filter.outputs.driver_events_changed == 'true' && needs.paths-filter.outputs.driver_schema_changed == 'false'
+    steps:
+      - name: Check driver SCHEMA_VERSION
+        uses: mshick/add-pr-comment@v2
+        with:
+          message: |
```
Comment (FedeDP) [inline .github/workflows/ci.yml:418]: I am not sure whether to make the job fail too (`exit 1`)

**Comment by Andreagit97** [comment]
the idea here is smart the only issue is that touching these files doesn't mean that we really need a versions bump :/
Just consider the `driver/bpf/fillers.h` file when you have just to rewrite a little piece of logic because the verifier is broken, or the `event_table` when you just have to add an `EF_OLD_VERSION`
Not sure what is the best way to avoid tons of false positives :/

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -391,3 +404,29 @@ jobs:
...
+    permissions:
+      pull-requests: write
+    needs: paths-filter
+    if: needs.paths-filter.outputs.driver_fillers_changed == 'true' && needs.paths-filter.outputs.driver_api_changed == 'false'
+    steps:
+      - name: Check driver API_VERSION
+        uses: mshick/add-pr-comment@v2
+        with:
+          message: |
+            Please double check **driver/API_VERSION** file. See [versioning](https://github.com/falcosecurity/libs/blob/master/driver/README.VERSION.md#api-version-number).
```
Comment (Molter73) [inline .github/workflows/ci.yml:432]: How about we move these to a separate workflow and have that run on the changes to the specific files? Seems they are mostly independent and changes to files that are not the 4 new ones defined at the top of the file have no reason to run any of these checks

**Review by Molter73 — COMMENTED** [review]
This is awesome! Thank you very much @FedeDP!!

**Comment by FedeDP** [comment]
>  Not sure what is the best way to avoid tons of false positives :/

Yep that is obvious; in fact the new job just ask user to double check if anything is needed. We cannot avoid false postiives.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Molter73) [reply .github/workflows/ci.yml:418]: I think the comment is enough, at the very least it should be a red flag for maintainers on its own.

**Review by Molter73 — APPROVED** [review]
/lgtm

**Review by leogr — APPROVED** [review]
:heart: 
LGTM.

However, I would like to still investigate if we can think more sophisticated way to handle this :astronaut:

**Pull request merged**